catalyst_voices_packages/catalyst_cardano_serialization/lib/src/transaction_output.dart

```diff
@@ -0,0 +1,363 @@
+import 'dart:typed_data';
+import 'package:catalyst_cardano_serialization/src/address.dart';
+import 'package:catalyst_cardano_serialization/src/scripts.dart';
+import 'package:catalyst_cardano_serialization/src/types.dart';
+import 'package:cbor/cbor.dart';
+import 'package:equatable/equatable.dart';
+import 'package:pinenacl/api.dart';
+
+/// Enum representing the type of a datum option in Cardano.
+/// A DatumOption can either be a datum hash or inline Plutus data.
+enum DatumOptionType {
+  /// [datumHash]: Represents a hash of the datum, typically used to refer to
+  /// data that exists off-chain.
+  datumHash(0),
+
+  /// [data]: Represents actual inline datum stored on-chain.
+  data(1);
+
+  /// The value of the enum.
+  final int value;
+
+  const DatumOptionType(this.value);
+}
+
+/// Abstract base class representing a datum in a Cardano transaction output.
+/// A datum is a piece of data that can be either a hash or inline plutus data.
+///
+/// Implementations of this class must provide a way to encode the datum into
+/// a CBOR.
+sealed class Datum extends Equatable implements CborEncodable {
```

Comment:
Consider to extract all Datum-related classes to a separate file, i.e. you can place it next to transaction_output.dart in a file name `datum.dart`. This `transaction_output.dart` file is primarily for the Transaction output and at the very least we could expect to see at the top of the file the TransactionOutput class. The structures that TransactionOutput consists of should be below (to allow top-to-bottom reading) or in separate file (preferred).

Reply:
Yeah, I have already have a `plutus.dart` in my local branch.

Reply:
Done